### ISSUE TITLE
Function argument are not available at the first line when debugging w/ GDB

### ISSUE THREAD

**Issue by PokhodenkoSA** [numba/numba#7415]
## Reporting a bug

- [x] I have tried using the latest released version of Numba (most recent is
 visible in the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as 'python bug.py'.

Python code:

```python
from numba import njit

@njit(debug=True)
def sum(a, b):
    c = a + b  # break here
    return c, a

print(sum(1, 2))
```

Annotations:

```python-traceback
$ NUMBA_DUMP_ANNOTATION=1 python chk_args_vals.py
-----------------------------------ANNOTATION-----------------------------------
# File: chk_args_vals.py
# --- LINE 4 --- 

@njit(debug=True)

# --- LINE 5 --- 

def sum(a, b):

    # --- LINE 6 --- 
    # label 0
    #   a = arg(0, name=a)  :: int64
    #   b = arg(1, name=b)  :: int64
    #   c = a + b  :: int64
    #   del b

    c = a + b

    # --- LINE 7 --- 
    #   $14build_tuple.5 = build_tuple(items=[Var(c, chk_info_args.py:6), Var(a, chk_info_args.py:6)])  :: UniTuple(int64 x 2)
    #   del c
    #   del a
    #   $16return_value.6 = cast(value=$14build_tuple.5)  :: UniTuple(int64 x 2)
    #   del $14build_tuple.5
    #   return $16return_value.6

    return c, a


================================================================================
(3, 1)
```

Debug session:

```bash
$ NUMBA_OPT=0 gdb -q --args python chk_args_vals.py
(gdb) break chk_args_vals.py:6
(gdb) run
...
Thread 1 "python" hit Breakpoint 1, __main__::sum$241 () at chk_args_vals.py:6
6           c = a + b
(gdb) info locals
a = 0
b = 0
c = 0
(gdb) n
7           return c, a
(gdb) info locals
a = 1
b = 0
c = 3
```

It seems that when breakpoints hit the first line of function then arguments are not available yet.
Enabling this scenario is very useful for setting conditional breakpoints depending on function arguments.

**Comment by stuartarchibald** [comment]
Thanks for the report. I've a couple of ideas as to why this is happening, will start looking.

**Comment by stuartarchibald** [comment]
#7177 fixes. Example under #7177, conditional break point based on arg to function from induction variable.
```
$ cat issue7415_2.py 
from numba import njit
import numpy as np

@njit(debug=True)
def sum(a, b):
    c = a + b  # break-here
    return c

@njit(debug=True)
def call_sum(a):
    acc = 0
    for i in range(10):
        print("i = ", i)
        acc += sum(i, a)[0]
    return acc

call_sum(np.ones(4))

$ NUMBA_OPT=0 NUMBA_EXTEND_VARIABLE_LIFETIMES=1 gdb -q --args python issue7415_2.py 
(gdb) break issue7415_2.py:6 if a == 4
No source file named issue7415_2.py.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (issue7415_2.py:6 if a == 4) pending.
(gdb) r
i =  0
i =  1
i =  2
i =  3
i =  4

Breakpoint 1, __main__::sum$242 (a=4, b=...) at issue7415_2.py:6
6           c = a + b  # break-here
(gdb) info args
a = 4
b = {meminfo = 0x5555566e3f80 "\003", parent = 0x7fffdd8e7c90 "\004", nitems = 4, itemsize = 8, data = 0x555556250750, 
  shape = {4}, strides = {8}}
(gdb) info locals
c = {meminfo = 0x0, parent = 0x0, nitems = 0, itemsize = 0, data = 0x0, shape = {0}, strides = {0}}
(gdb) where
#0  __main__::sum$242 (a=4, b=...) at issue7415_2.py:6
...
```

**Comment by stuartarchibald** [comment]
Closed by #7177.
